### PR TITLE
fix conflict between Path("/**") and PathSubtree("/")

### PR DESCRIPTION
**Pull request by aryszka**
fix conflict between Path("/**") and PathSubtree("/")

Fixes: #1198 

Signed-off-by: Arpad Ryszka <arpad.ryszka@gmail.com>

### CODE REVIEW

File: routing/matcher.go [hunk 1]
```diff
@@ -391,19 +418,17 @@ func matchPathTree(tree *pathmux.Tree, path string, lrm *leafRequestMatcher) (ma
 		return nil, nil
 	}
 
+	fmt.Println("params found", params)
```
Comment (szuecs) [inline routing/matcher.go]: this is a left over?

Comment (aryszka) [reply routing/matcher.go]: indeed it was

File: routing/matcher.go [hunk 2]
```diff
@@ -283,15 +290,85 @@ func moveToSubtreeIfExists(subtree *pathMatcher, paths map[string]*pathMatcher,
...
+	for _, p := range removeFromPaths {
+		delete(paths, p)
+	}
+}
+
 func moveConflictingToSubtree(subtrees, paths map[string]*pathMatcher) {
 	for p, stm := range subtrees {
 		moveToSubtreeIfExists(stm, paths, p)
+
+		// TODO: this may mean that a non-subtree route will match even when ignore-trailing-match is
```
Comment (szuecs) [inline routing/matcher.go]: non-subtree route is a route having `Path` and not `PathSubtree` ?
Do you want to fix this in this PR?

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Pull request merged**